include/boost/url/url_base.hpp

```diff
@@ -35,6 +35,7 @@ struct any_params_iter;
 struct any_segments_iter;
 struct params_iter_impl;
 struct segments_iter_impl;
+struct url_template;
```

Comment:
maybe we can rename `url_template` to `pattern`?

Reply:
I'm fine with both. `pattern` seems a little generalistic but I'm assuming you don't like `url_pattern`. It's an implementation detail anyway so we can change that as much as we want.